### PR TITLE
feat(community): Remove required param from LanceDB integration

### PR DESCRIPTION
**Pull request by AyushExel**
Hello. I'm from LanceDB team. The integration used old version of lancedb and user had to create a table with dummy data in order to ingest stuff. This makes passing Table optional. If not found, the integration will automatically create table if not passed by the user, making it easier to work with LanceDB<>Langchain.

### CODE REVIEW

**Comment by AyushExel** [comment]
Hey @jacoblee93 can you please take a look at this? Its been requested to improve UX from our community users. Thanks in advance!

**Comment by jacoblee93** [comment]
Apologies for the delay, having a look now!

File: libs/langchain-community/src/vectorstores/lancedb.ts [hunk 1]
```diff
@@ -18,15 +21,24 @@ export type LanceDBArgs = {
...
-    this.table = args.table;
+  private uri: string;
+
+  private table_name: string;
+
+  private mode: string;
+
+  constructor(embeddings: EmbeddingsInterface, args?: LanceDBArgs) {
+    super(embeddings, args || {});
+    this.table = args?.table || undefined;
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/lancedb.ts:36]: nit: `undefined` is implicit here

File: libs/langchain-community/src/vectorstores/lancedb.ts [hunk 2]
```diff
@@ -8,8 +8,11 @@ import { Document } from "@langchain/core/documents";
  * table and an optional textKey.
  */
 export type LanceDBArgs = {
-  table: Table;
+  table?: Table;
   textKey?: string;
+  uri?: string;
+  table_name?: string;
```
Comment (jacoblee93) [inline libs/langchain-community/src/vectorstores/lancedb.ts:14]: nit: Use camel case for consistency

**Comment by jacoblee93** [comment]
Thanks for this!

**Review by jacoblee93 — APPROVED** [review]

**Comment by jacoblee93** [comment]
Hey @AyushExel, this is failing CI:

```bash
Error: src/vectorstores/lancedb.ts(89,9): error TS2769: No overload matches this call.
  Overload 1 of 5, '(name: string, data: Record<string, unknown>[] | Table<any>, options: WriteOptions): Promise<Table<number[]>>', gave the following error.
    Type 'string' is not assignable to type 'WriteMode | undefined'.
  Overload 2 of 5, '(name: string, data: Record<string, unknown>[] | Table<any>, embeddings: EmbeddingFunction<number[]>): Promise<Table<number[]>>', gave the following error.
    Argument of type '{ writeMode: string; }' is not assignable to parameter of type 'EmbeddingFunction<number[]>'.
      Object literal may only specify known properties, and 'writeMode' does not exist in type 'EmbeddingFunction<number[]>'.
```

Could you have a look?

**Comment by AyushExel** [comment]
Updated! Can you take a look again? Thanks!

**Comment by jacoblee93** [comment]
Thank you!

**Comment by jacoblee93** [comment]
Still getting a few:

```bash
Error: src/indexes/vector_stores/lancedb/fromDocs.ts(37,7): error TS2740: Type 'Table<number[]>' is missing the following properties from type 'Table<number[]>': createScalarIndex, countRows, delete, update, and 9 more.
Error: src/indexes/vector_stores/lancedb/fromTexts.ts(31,7): error TS2322: Type 'import("/home/runner/work/langchainjs/langchainjs/examples/node_modules/vectordb/dist/index").Table<number[]>' is not assignable to type 'import("/home/runner/work/langchainjs/langchainjs/node_modules/vectordb/dist/index").Table<number[]>'.
Error: src/indexes/vector_stores/lancedb/load.ts(17,61): error TS2322: Type 'import("/home/runner/work/langchainjs/langchainjs/examples/node_modules/vectordb/dist/index").Table<number[]>' is not assignable to type 'import("/home/runner/work/langchainjs/langchainjs/node_modules/vectordb/dist/index").Table<number[]>'.
Error:  command finished with error: command (/home/runner/work/langchainjs/langchainjs/examples) /tmp/xfs-3a2609fc/yarn run build exited (2)
Error: examples#build: command (/home/runner/work/langchainjs/langchainjs/examples) /tmp/xfs-3a2609fc/yarn run build exited (2)
```

I can try to have a look later too - or you can run `yarn build` from within `@langchain/community` to build it locally

**Comment by AyushExel** [comment]
@jacoblee93 could you point me to where this example is in the repo? I'm able to run stuff locally without error so not sure what's happening

**Comment by AyushExel** [comment]
@jacoblee93 okay tested the examples folder locally. Needed to update lancedb package version there as well. should be working now

**Comment by jacoblee93** [comment]
Hey sorry got super busy with releasing everything for 0.3 - will have a look at this today!

**Comment by jacoblee93** [comment]
Thank you for your patience! Will ship this tonight.

**Comment by jacoblee93** [comment]
Live in community@0.3.1!

**Comment by AyushExel** [comment]
Awesome thanks!

**Pull request merged**